feats/redis.py

```diff
@@ -0,0 +1,152 @@
+from typing import List
+from redis import Redis
+from redis.exceptions import ConnectionError
+
+
+class NotConnectedError(ConnectionError):
+    pass
+
+
+class StreamIterator:
+    """
+    An iterator class used to iterate over the entirety of a feature stream
+    This works by pre-loading the entire stream into memory
+    Note: any updates to the stream during iteration will be left out
+    eg if a value is written in the middle of iteration it will not appear as
+    the last value at the end of the loop
+    """
+    def __init__(self, stream):
+        self.index = 0
+        self.end = len(stream)
+        self.stream = stream
+
+    def __iter__(self):
+        self.index = 0
+        return self
+
+    def __next__(self):
+        if self.index >= self.end:
+            raise StopIteration
+        result = self.stream[self.index]
+        self.index += 1
+        return result[1]
+
+
+class FeatureStream:
+    """
+    This class provides utility methods for retrieving data from Redis.
+    It should not be initialized directly but instead returned from keying off
+    a RedisClient instance.
+    """
+    def __init__(self, redis, key):
+        self.key = key
```

Comment:
We might want to namespace the keys given here in case we want to store things other than features in the same db. The IP had `feature:${name}` as the key, but the actual impl doesn't matter much.